src/main/scala/is/hail/stats/Eigendecomposition.scala

```diff
@@ -34,4 +37,14 @@ case class Eigendecomposition(rowSignature: Type, rowIds: Array[Annotation], eve
   def evectsSpark(): linalg.DenseMatrix = new linalg.DenseMatrix(evects.rows, evects.cols, evects.data, evects.isTranspose)
   
   def evalsArray(): Array[Double] = evals.toArray
+  
+  def distribute(sc: SparkContext): EigendecompositionDist = {
+    val U = BlockMatrixIsDistributedMatrix.from(sc, evects.asSpark(), 1024, 1024)
+    EigendecompositionDist(rowSignature, rowIds, U, evals)
+  }
+}
+
+case class EigendecompositionDist(rowSignature: Type, rowIds: Array[Annotation], evects: BlockMatrix, evals: DenseVector[Double]) {
+  require(evects.numRows() == rowIds.length) // FIXME: expensive
```

Comment:
Yes, and it looks like same is true after BlockMatrix mult, so I've removed the FIXME